sqlglot/optimizer/qualify_columns.py

```diff
@@ -122,6 +123,30 @@ def _expand_using(scope, schema):
                 scope.replace(column, replacement)
 
 
+def _expand_group_numbers(scope):
+    selects = scope.expression.args.get("expressions")
+    group = scope.expression.args.get("group")
+    if not group:
+        return
+
+    new_expressions = []
+    for expression in group.args.get("expressions"):
+        if isinstance(expression, exp.Literal) and expression.is_int:
```

Comment:
do you mind refactoring so is_int and is_string is on base object so we don't always have to check if the expression is a literal?